packages/app-store/routing-forms/pages/route-builder/[...appPages].tsx

```diff
@@ -280,23 +282,29 @@ const Route = ({
                         }
                         if (option.value !== "custom") {
                           setRoute(route.id, { action: { ...route.action, value: option.value } });
+                          setCustomEventTypeSlug("");
                         } else {
                           setRoute(route.id, { action: { ...route.action, value: "custom" } });
                           setCustomEventTypeSlug("");
                         }
                       }}
                       value={
                         eventOptions.length !== 0 && route.action.value !== ""
-                          ? eventOptions.find((eventOption) => eventOption.value === route.action.value) || {
+                          ? eventOptions.find(
+                              (eventOption) =>
+                                eventOption.value === route.action.value && !customEventTypeSlug.length
```

Comment:
fixes issue that when `customEventTypeSlug` is same as an existing event type slug the existing event type was selected in the select instead of 'custom'